gnocchi/indexer/sqlalchemy.py

```diff
@@ -1104,7 +1105,18 @@ def list_resources(self, resource_type='generic',
                                    != pymysql.constants.ER.NO_SUCH_TABLE)):
                                 raise
 
-            return all_resources
+            if detailed_metrics:
+                return list(map(self._load_detailed_metrics, all_resources))
+            else:
+                return all_resources
+
+    @staticmethod
+    def _load_detailed_metrics(r):
+        # NOTE(sileht): This moves the r in metric.resource without
+        # querying the database
```

Comment:
just a question, but this is true only becasue we use joinedload above right?

Reply:
yes